### DISCUSSION TITLE
HL7 2.3.1 ORM01 - Scheduled Start Date/Time, name and AET not set

### DISCUSSION THREAD

**Discussion by fonteneu** [dcm4che/dcm4chee-arc-light#4604] [Q&A] [answered]
An external HL7 2.3.1 server set these fields

These work fine in dcm4chee 2.x, but not in arc light.

* ORC.7.4: start date/time.  Arc Light errors with `SPS Date / Time is missing in ORC^1^7^1^4 - use current date time`. Attempting to set ORC.1.7.1.4 leads to the same error.
* PID 5.1 and 5.2 - name. This ends up empty in (1010,1010) for Arc Light.
* ZDS.2.1 - Scheduled Station AE. This ends up with some default "SCHEDULEDSTATION" in (0040,0001) rather than the actual AET.

Any tips to fix these would be greatly appreciated

**Comment by vrindanayak** [comment]
> An external HL7 2.3.1 server set these fields
> 
> These work fine in dcm4chee 2.x, but not in arc light.
> 
>     * ORC.7.4: start date/time.  Arc Light errors with `SPS Date / Time is missing in ORC^1^7^1^4 - use current date time`. Attempting to set ORC.1.7.1.4 leads to the same error.

`use current date time` is not an Error, but is provided as an INFO message in the logs. The SPS Date Time are supported as part of TQ1.7 for Imaging Order messages OMI^O23. For ORM^O01, it seems that the support for ORC.7.4 got lost in some refactoring, will add / fix it

>     * PID 5.1 and 5.2 - name. This ends up empty in (1010,1010) for Arc Light.

Patient's Name tag is referenced by (0010,0010) not (1010,1010). Can you share the sample HL7 message or the PID.5 value to check further?
>     * ZDS.2.1 - Scheduled Station AE. This ends up with some default "SCHEDULEDSTATION" in (0040,0001) rather than the actual AET.

ZDS is a custom segment according to IHE RAD-4 and only field 1 of this segment is supported / mentioned in the standard - https://www.ihe.net/uploadedFiles/Documents/Radiology/IHE_RAD_TF_Vol2.pdf#page=115

**Reply by fonteneu** [reply]
ORC.7.4: thanks for fixing, will the Docker image be updated as well?

ZDS.2.1: This works in dcm4chee 2.x, not sure why. 

Which HL7 field should be set to specify Scheduled AET ?

PID 5.1 and 5.2: Yes, I meant 0010,0010. But it's empty. I'll see if I can provide an example later.

**Reply by vrindanayak** [reply]
> ORC.7.4: thanks for fixing, will the Docker image be updated as well?

Yes, I will open an issue and reference it here. Whenever it is fixed and an archive image is released, it will be part of that dockerized archive version.

> ZDS.2.1: This works in dcm4chee 2.x, not sure why.
> 
> Which HL7 field should be set to specify Scheduled AET ?

Custom dcm4chee archive 5.x solution for Scheduled Station AET in ORM^O01 is supported as part of ORC.18 

> PID 5.1 and 5.2: Yes, I meant 0010,0010. But it's empty. I'll see if I can provide an example later.

Ok

**Comment by fonteneu** [comment]
ORC.18 works, thanks you very much. PID 5 (patient name) is also correct in 0010,0010 after closer inspection.

Do you think the ORC.7.4 fix for scheduled time and date is far away?

**Reply by vrindanayak** [reply, marked as answer]
> Do you think the ORC.7.4 fix for scheduled time and date is far away?

Opened https://github.com/dcm4che/dcm4chee-arc-light/issues/4612. Should be fixable in a day, but will be part of next archive version whenever it is released. You may use the fixed XSL and update your vendor data configuration, once this issue is closed.
